crates/compiler/mono/src/tail_recursion.rs

```diff
@@ -1098,9 +1102,6 @@ fn expr_contains_symbol(expr: &Expr, needle: Symbol) -> bool {
         }),
```

Comment:
Don't forget to update `Expr::Tag { arguments, .. } => arguments.contains(&needle),` to check if the reuse token contains the symbol.